test/e2e/framework/clean_up.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+
+	"sigs.k8s.io/cluster-api-provider-azure/test/e2e/framework/management/kind"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+)
+
+// CleanUpInput are all the dependencies needed to clean up a Cluster API cluster.
+type CleanUpInput struct {
+	Management    *kind.Cluster
+	Cluster       *clusterv1.Cluster
+	DeleteTimeout time.Duration
+}
+
+func (c *CleanUpInput) setDefaults() {
+	if c.DeleteTimeout == 0*time.Second {
```

Comment:
nit: spaces missing

Reply:
this `0*time.Second` is made by the go tooling, if that's what you mean.